adaptive/learner/average_learner.py

```diff
@@ -30,7 +32,9 @@ class AverageLearner(BaseLearner):
         Number of evaluated points.
     """
 
-    def __init__(self, function, atol=None, rtol=None):
+    def __init__(self, function, atol=None, rtol=None, min_npoints=2):
+        if min_npoints < 2:
+            raise ValueError("`min_npoints` should be at least 2.")
```

Comment:
Should we really raise or just sneakily

```suggestion
        # Cannot estimate standard deviation with less than 2 points.
        min_npoints = max(min_npoints, 2)
```

instead?

Reply:
I thought about that too but ultimately, I don't really have a preference.

Do you?

Reply:
I think what I proposed is nicer: it's not that we don't respect the user settings if the user asks for at least fewer points than 2. And we don't have to document it if there's no error.